### PR TITLE
pipeline reconciler with queue

### PR DESCRIPTION
**Pull request by sfwn**
#### What type of PR is this?

feature

#### What this PR does / why we need it:

- Pipeline add queue management
- Reconciler reconcile pipeline with queue

### CODE REVIEW

File: modules/pipeline/pipengine/reconciler/queuemanage/manager/pipeline.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+		return cachedP
+	}
+
+	// query from db
+	var p *spec.Pipeline
+	_ = loop.New(loop.WithDeclineLimit(time.Second*10), loop.WithDeclineRatio(2)).Do(func() (abort bool, err error) {
+		_p, exist, err := mgr.dbClient.GetPipelineWithExistInfo(pipelineID)
+		if err != nil {
+			err = fmt.Errorf("failed to query pipeline: %d, err: %v", pipelineID, err)
+			logrus.Error(err)
```
Comment (kakj-go) [inline modules/pipeline/pipengine/reconciler/queuemanage/manager/pipeline.go:103]: logrus.Errorf()  Is it better

File: modules/pipeline/pipengine/reconciler/queuemanage/queue/v_capacity.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+package queue
+
+import (
+	"fmt"
+
+	"github.com/erda-project/erda/pkg/strutil"
```
Comment (kakj-go) [inline modules/pipeline/pipengine/reconciler/queuemanage/queue/v_capacity.go:19]: import

File: modules/pipeline/services/pipelinesvc/create_v2.go [hunk 3]
```diff
@@ -292,7 +292,18 @@ func (s *PipelineSvc) makePipelineFromRequestV2(req *apistructs.PipelineCreateRe
 
 	// queue
 	if req.BindQueue != nil {
-		p.Extra.QueueID = req.BindQueue.ID
+		customPriority := req.BindQueue.Priority
+		customPriorityStr, ok := p.MergeLabels()[apistructs.LabelBindPipelineQueueCustomPriority]
+		if ok {
+			_customPriority, err := strconv.ParseInt(customPriorityStr, 10, 64)
```
Comment (kakj-go) [inline modules/pipeline/services/pipelinesvc/create_v2.go:298]: _begin is not a good name

File: modules/pipeline/pipengine/reconciler/queuemanage/manager/pipeline.go (hunk 1 above)
Comment (sfwn) [reply modules/pipeline/pipengine/reconciler/queuemanage/manager/pipeline.go:103]: you are right. I removed some logic here which use pre-defined err, so left it alone.
i will polish it.

**Review by Effet — APPROVED** [review]

**Pull request merged**